packages/graphql/src/translate/cypher-builder/functions/PredicateFunctions.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { CypherEnvironment } from "../Environment";
+import type { Pattern } from "../Pattern";
+import { Where } from "../sub-clauses/Where";
+import type { Expr, Predicate } from "../types";
+import { compileCypherIfExists } from "../utils";
+import type { Variable } from "../variables/Variable";
+import { CypherFunction } from "./CypherFunction";
+
+export class PredicateFunction extends CypherFunction {}
+
+class PredicateFunctionWithPattern extends PredicateFunction {
```

Comment:
In my opinion, we could create a `ListPredicateFunction` rather than a `PredicateFunctionWithPattern`, these will include `any`, `some`, `none`, `single` and `all`.  All these can be written in an expression as `variable IN list WHERE predicate`

Reply:
Yep, I think you are right. 
For now, I'll rename this class, but keep only the `any` and `all` as we don't use the other ones for now